joomla/joomla-cms
# [4.0] Plg parentassociations

Pull Request for Issue # .

### Summary of Changes
Implement a color presentation in which one can see if the same articles are in the same state in different languages. The user has to manually check as articles that are not actually marked, adjust if necessary, and finally confirm as current.
If there are several articles in different languages with the same content, they can already be linked to each other in order to identify their belonging together. In order to simplify the above-mentioned functionality, a master-slave relationship has also been implemented between the articles. Specifically, this means that within an article group an article exists which is defined as the main article (master). All related articles should derive from the main article and from their slaves. In the application, all slaves are marked as unactual as soon as the master changes.


### Testing Instructions
1. Install at least 4 (e.g., De, ID, EN, FR, AF) languages
When you install a language, you must add the following values ​​in the language as in  C:\xampp1\htdocs\WPW3\administrator\language\en-GB\en-GB.com_associations.ini 
add the following values ​​in the appropriate language.
COM_ASSOCIATIONS_SAVE_APPROVED="Save Approved"
COM_ASSOCIATIONS_NOT_APPROVED_NOPARENT="reference is not the master"
COM_ASSOCIATIONS_MESSAGE_APPROVED="approved"
COM_ASSOCIATIONS_NOTCHILD="firstly save the Target Article 

2. Create  first article (DE)
3. Create second article (ID)
4. Create a third article (EN)
5. First article with second associate
6. Compare languages and press "Save Approved" as shown in Figure 1
7. First article with third article associate
8. In the Component Multilingual Associtation, select the first article as shown in Figure 2
9. Select third article, red exclamation point should appear according to main language (Figure 3)
10. If item is green, then in the info text Approved = 1 (see first picture).
11. If articles are yellow, then in the info text Approved = 0.
12. If articles are gray, then in the info text without Approved



### Expected result
Figure 1:
![test3](https://user-images.githubusercontent.com/26348675/30077458-2f8c338c-927c-11e7-81d7-076f6ab29e2d.png)
Figure 2:
![test](https://user-images.githubusercontent.com/26348675/30077496-50217170-927c-11e7-8404-126bc72349a0.png)
Figure 3:
![test2](https://user-images.githubusercontent.com/26348675/30077504-576cd6b8-927c-11e7-8a1f-c4e414db7b55.png)
